app/src/org/commcare/models/database/user/models/EntityStorageCache.java

```diff
@@ -133,11 +135,17 @@ public static int getSortFieldIdFromCacheKey(String detailId, String cacheKey) {
         }
     }
 
-    public static void tryWipeCache() {
+    public static void wipeCacheForCurrentApp() {
         SQLiteDatabase userDb = CommCareApplication.instance().getUserDbHandle();
-        SqlStorage.wipeTable(userDb, TABLE_NAME);
-        String uuid = CommCareApplication.instance().getSession().getLoggedInUser().getUniqueId();
-        setEntityCacheWipedPref(uuid, CommCareApplication.instance().getCurrentApp().getAppRecord().getVersionNumber());
+        userDb.beginTransaction();
+        try {
+            userDb.delete(TABLE_NAME, COL_APP_ID + " = ?", new String[]{ReportingUtils.getAppId()});
```

Comment:
I'm pretty uncomfortable with getting the current app ID from ReportingUtils. I think we should move that somewhere more semantically "important", since I wouldn't anticipate someone making changes to ReportingUtils assuming they would break core app behavior if they made changes.